File: Java-SpringBoot/BE-01-InMemory/java-21-springboot-4.0/src/main/java/com/flyrank/userservice/dto/UserCreateRequest.java
package com.flyrank.userservice.dto;

import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.validation.Valid;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotNull;

/**
 * Payload accepted by POST /user. Only name is mandatory.
 */
@Schema(description = "User creation request payload")
public record UserCreateRequest(
        @Schema(description = "User's first and last name (mandatory)", requiredMode = Schema.RequiredMode.REQUIRED)
        @NotNull(message = "name object is required")
        @Valid
        NameDto name,

        @Schema(description = "Email address (optional)", example = "jane.doe@example.com")
        @Email(message = "value is not a valid email address")
        String email,

        @Schema(description = "Telephone number (optional)", example = "+1-555-0199")
        String telephone
) {}

File: Java-SpringBoot/BE-01-InMemory/kotlin-2.3-springboot-3.5/src/main/kotlin/com/flyrank/userservice/dto/UserCreateRequest.kt
package com.flyrank.userservice.dto

import io.swagger.v3.oas.annotations.media.Schema
import jakarta.validation.Valid
import jakarta.validation.constraints.Email
import jakarta.validation.constraints.NotNull

/**
 * Payload accepted by POST /user. Only name is mandatory.
 */
@Schema(description = "User creation request payload")
data class UserCreateRequest(
    @field:Schema(description = "User's first and last name (mandatory)", requiredMode = Schema.RequiredMode.REQUIRED)
    @field:NotNull(message = "name object is required")
    @field:Valid
    val name: NameDto?,

    @field:Schema(description = "Email address (optional)", example = "jane.doe@example.com")
    @field:Email(message = "value is not a valid email address")
    val email: String? = null,

    @field:Schema(description = "Telephone number (optional)", example = "+1-555-0199")
    val telephone: String? = null
)
